eng/common/Update-Change-Log.ps1

```diff
@@ -0,0 +1,135 @@
+﻿# Note: This script will add or replace version title in change log
+param (
+  [Parameter(Mandatory = $true)]
+  [String]$Version,
+  [Parameter(Mandatory = $true)]
+  [String]$ChangeLogPath,
+  [String]$Unreleased = $True,
+  [String]$ReplaceVersion = $False
+)
+
+# Parameter description
+# Version : Version to add or replace in change log
+# ChangeLogPath: Path to change log file. If change log path is set to directory then script will probe for change log file in that path 
+# Unreleased: Default is true. If it is set to false, then today's date will be set in verion title. If it is True then title will show "Unreleased"
+# ReplaceVersion: This is useful when replacing current version title with new title.( Helpful to update the title before package release)
+
+$RELEASE_TITLE_REGEX = "(?<releaseNoteTitle>^\#+.*(?<version>\b\d+\.\d+\.\d+([^0-9\s][^\s:]+)?))"
+
+function Version-Matches($line)
+{
+    return ($line -match $RELEASE_TITLE_REGEX)
+}
+
+function Get-ChangelogPath
+{
+  Param (
+        [String]$Path
+   )
+  
+   # Check if CHANGELOG.md is present in path 
+   $ChangeLogPath = Join-Path -Path $Path -ChildPath "CHANGELOG.md"
+   if ((Test-Path -Path $ChangeLogPath) -eq $False){
+      # Check if change log exists with name HISTORY.md
+      $ChangeLogPath = Join-Path -Path $Path -ChildPath "HISTORY.md"
+      if ((Test-Path -Path $ChangeLogPath) -eq $False){
+         Write-Host "Change log is not found in path[$Path]"
+         exit(1)
+      }
+   }
+  
+   Write-Host "Change log is found at path [$ChangeLogPath]"
+   return $ChangeLogPath
+}
+
+
+function Get-VersionTitle
+{
+   Param (
+        [String]$Version,
+        [String]$Unreleased
+   )
+   # Generate version title
+   $newVersionTitle = "## $Version (Unreleased)"
+   if ($Unreleased -eq $False){
+      $releaseDate = Get-Date -Format "(yyyy-MM-dd)"
+      $newVersionTitle = "## $Version $releaseDate"
+   }
+   return $newVersionTitle
+}
+
+
+function Get-NewChangeLog
+{
+   Param (
```

Comment:
minor: For functions you don't need to specify the parameters in this long hand way. You can make them inline like:
```
function Get-VersionTitle([String]$Version, [String]$Unreleased) {
```

You also don't have to explicitly provide the type if  it isn't interesting:

```
function Get-VersionTitle($Version, $Unreleased) {
```